runtime/src/accounts.rs

```diff
@@ -320,36 +332,88 @@ impl Accounts {
     /// Slow because lock is held for 1 operation instead of many
     pub fn store_slow(&self, fork: Fork, pubkey: &Pubkey, account: &Account) {
         let mut accounts = HashMap::new();
-        accounts.insert(pubkey, (account, 0));
+        accounts.insert(pubkey, account);
         self.accounts_db.store(fork, &accounts);
     }
 
     fn lock_account(
         locks: &mut HashSet<Pubkey>,
-        keys: &[&Pubkey],
+        credit_only_locks: &Arc<RwLock<HashMap<Pubkey, CreditOnlyLock>>>,
+        message: &Message,
         error_counters: &mut ErrorCounters,
     ) -> Result<()> {
-        // Copy all the accounts
-        for k in keys {
+        let (credit_debit_keys, credit_only_keys) = message.get_account_keys_by_lock_type();
+
+        for k in credit_debit_keys.iter() {
+            let credit_only_locks = credit_only_locks.read().unwrap();
+            if locks.contains(k)
+                || (credit_only_locks.contains_key(k)
```

Comment:
is this pre-check faster than get(&k).and_then() or equivalent?

Reply:
I don't know; I honestly didn't think of using Option methods. `map_or` certainly looks cleaner.